snmp/datadog_checks/snmp/snmp.py

```diff
@@ -98,13 +106,49 @@ def _get_instance_key(self, instance):
 
         return key
 
+    def discover_instances(self):
+        network = ipaddress.ip_network(self._config.network_address)
+        discovery_interval = self._config.instance.get('discovery_interval', 3600)
+        discovery_pause = self._config.instance.get('discovery_pause', 1)
+        while self._running:
+            start_time = time.time()
+            for host in network.hosts():
```

Comment:
Should we empty `self._config.discovered_instances` before `for host in network.hosts()` ? Otherwise, if some network hosts that does not exist anymore will be kept in the list indefinitely.

Reply:
So, no, we can't clean it up, because I don't want to restart from scratch every time. I think I'm tempted to leave it for later: we should add an option like remove an host if it fails for X times.

Reply:
Do you mind adding a `TODO` about this ?

But ideally, that should something we should handle now since having something that grows over time it not great.

Reply:
It doesn't magically grow over time, you need to install physical machines. I added a todo.